app/Http/Requests/TwilioRequest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Http\Requests\Request;
+use App\PhoneNumber;
+use App\Friend;
+use Illuminate\Foundation\Validation\ValidatesRequests;
+use Services_Twilio_RequestValidator;
+
+class TwilioRequest extends Request
+{
+    /**
+     * @var PhoneNumber The model instance retrieved using the From number.
+     */
+    private $phoneNumber = null;
+
+    /**
+     * Validate that the incoming request is from twilio.
+     *
+     * @return bool
+     */
+    public function authorize(Services_Twilio_RequestValidator $validator)
+    {
+        return app()->isLocal() || $validator->validate(
+            $this->headers->get('X-Twilio-Signature'),
+            $this->fullUrl(),
+            $this->input()
+        );
+    }
+
+    public function rules()
+    {
+        return [];
+    }
+
+    public function phoneNumber()
+    {
+        if (is_null($this->phoneNumber)) {
+            //set the number to first [PhoneNumber | Friend | NULL]
+            $this->phoneNumber = PhoneNumber::byNumber($this->input('From'))->first()
```

Comment:
I don't think it will since `PhoneNumber` is checked before `Friend` but I'll have to add a test to be sure.

That does bring up another issue, it doesn't look like I added a way for the controller to restrict to `PhoneNumber` when accepting calls.